zhaquirks/philips/__init__.py

```diff
@@ -75,6 +98,34 @@ async def bind(self):
         return result
 
 
+class ButtonPressQueue:
+    """Philips button queue to derive multiple press events."""
+
+    def __init__(self):
+        """Init."""
+        self._ms_threshold = 500
+        self._ms_last_click = 0
+        self._click_counter = 1
+        self._callback = lambda x: None
+        self._task = asyncio.ensure_future(self._job())
+
+    async def _job(self):
+        await asyncio.sleep(self._ms_threshold / 1000)
+        self._callback(self._click_counter)
+
+    def press(self, callback):
+        """Process a button press."""
+        self._callback = callback
+        now_ms = time.time() * 1000
+        if now_ms - self._ms_last_click > self._ms_threshold:
+            self._click_counter = 1
+        else:
+            self._task.cancel()
```

Comment:
Just to confirm, there's no chance we could get here and task be already done?

Reply:
No I don't believe so.